robotoff/prediction/ocr/grammar.py

```diff
@@ -0,0 +1,155 @@
+        ]
+
+    return output
+
+
+def generate_terminal_symbols_file(
+    output_path: Path,
+    lang: str,
+    terminal_name: str,
+    taxonomy_type: TaxonomyType,
+    terminal_priority: Optional[int] = None,
+    lowercase: bool = True,
+    strip_accent: bool = True,
+    add_synonyms: bool = True,
+    add_xx: bool = True,
+    ignore_ids: Optional[set[str]] = None,
+    raises: bool = False,
+):
+    """Generate lark terminal symbol file from a taxonomy."""
+    content, name_to_id_mapping = generate_terminal_symbols_text(
+        lang=lang,
+        terminal_name=terminal_name,
+        taxonomy_type=taxonomy_type,
+        terminal_priority=terminal_priority,
+        lowercase=lowercase,
+        strip_accent=strip_accent,
+        add_synonyms=add_synonyms,
+        add_xx=add_xx,
+        ignore_ids=ignore_ids,
+        raises=raises,
+    )
+
+    with output_path.open("w") as f:
+        f.write("// This file has been generated automatically, DO NOT EDIT!\n")
+        f.write(content)
+
+    map_path = output_path.parent / f"{output_path.stem}_map.json"
+    dump_json(map_path, name_to_id_mapping)
+
+
+def generate_terminal_symbols_text(
+    lang: str,
+    terminal_name: str,
+    taxonomy_type: TaxonomyType,
+    terminal_priority: Optional[int] = None,
+    lowercase: bool = True,
+    strip_accent: bool = True,
+    add_synonyms: bool = True,
+    add_xx: bool = True,
+    ignore_ids: Optional[set[str]] = None,
+    raises: bool = True,
+) -> tuple[str, dict[str, list[str]]]:
+    ignore_ids = ignore_ids or set()
+    texts = []
+    taxonomy = get_taxonomy(taxonomy_type.name)
+    seen_set: dict[str, str] = {}
+    previous_node_id = None
+
+    node_id_names = extract_taxonomy_names(
+        lang, taxonomy, ignore_ids, add_synonyms, add_xx, lowercase, strip_accent
+    )
+    for node_id, name in node_id_names:
+        if node_id != previous_node_id:
+            if name in seen_set:
+                error_message = f"item {name} (node {node_id}) already seen for node {seen_set[name]}!"
+                if raises:
+                    raise ValueError(error_message)
+                else:
+                    logger.warning(error_message)
+            previous_node_id = node_id
+        seen_set[name] = node_id
+
+    first_node = True
+    name_to_id_mapping: dict[str, list[str]] = {}
+    for node_id, name in sorted(node_id_names, key=lambda x: len(x[1]), reverse=True):
+        name_to_id_mapping.setdefault(name, []).append(node_id)
+        if first_node:
+            terminal_priority_str = (
+                terminal_priority if terminal_priority is not None else ""
+            )
+            item = f"{terminal_name}.{terminal_priority_str}: "
+            first_node = False
+            len_first_item = len(item) - 2
+        else:
+            item = f"{' ' * len_first_item}| "
+
+        name = name.replace("/", "\\/")
+        item += f'/(?<!\\w){name}(?!\\w)/  // "{node_id}"'
```

Comment:
Don't you want to use word boundaries instead of look ahead and look behind which can be expensive.

```suggestion
        item += f'/\b{name}\b/  // "{node_id}"'
```

Reply:
I didn't know about `\b`, that's a nice trick, thanks!